packages/cli/src/prebuild.ts

```diff
@@ -638,6 +639,29 @@ ${utilsExport}
     `
   );
 
+  const redirects = siteData.build.pages?.redirects;
+  if (redirects !== undefined && redirects.length > 0) {
+    spinner.text = "Generating redirects";
+
+    for (const redirect of redirects) {
+      const redirectPagePath = generateRemixRoute(redirect.old);
+      const redirectFileName = `${redirectPagePath}.ts`;
+
+      const redirectPath = matchPathFromRedirects(redirect.old, redirects);
```

Comment:
I don't get what this does? Why we match old path with redirects when you already have an access to redirect.new? Remix shouldn't be involved for sure. Our dynamic path syntax is not 100% compatible with remix router.